tools/test.py

```diff
@@ -104,7 +104,10 @@ def main():
     # load the model of the backend
     model = task_processor.init_backend_model(args.model)
 
-    device_id = parse_device_id(args.device)
+    if args.device == 'cpu':
+        device_id = 0
+    else:
+        device_id = parse_device_id(args.device)
```

Comment:
Seems `device_id` get set to 0 for both `cpu` and `cuda:0` device. Maybe we have pass None to device_ids in MMDataParallel like described [here](https://github.com/open-mmlab/mmcv/blob/b57825f3f1f38cf6033e0d6d36af644cc57465fa/mmcv/parallel/data_parallel.py#L28). Besides, `with_sync` also need to be taken into account.

Reply:
Thanks, fixed.